File: conditionwatcher/src/main/java/com/azimolabs/conditionwatcher/Instruction.java
package com.azimolabs.conditionwatcher;

import android.os.Bundle;

/**
 * Created by F1sherKK on 16/12/15.
 */
public abstract class Instruction {

    private Bundle dataContainer = new Bundle();

    public final void setData(Bundle dataContainer) {
        this.dataContainer = dataContainer;
    }

    public final Bundle getDataContainer() {
        return dataContainer;
    }

    public abstract String getDescription();

    public abstract boolean checkCondition();
}

File: conditionwatcher/src/main/java/com/azimolabs/conditionwatcher/Instruction.kt
package com.azimolabs.conditionwatcher

import android.os.Bundle

/**
 * Created by F1sherKK on 16/12/15.
 */
abstract class Instruction {
    var dataContainer = Bundle()
        private set

    fun setData(dataContainer: Bundle) {
        this.dataContainer = dataContainer
    }

    abstract val description: String?
    abstract fun checkCondition(): Boolean
}